config-store-yaml/README.adoc

```diff
@@ -0,0 +1,74 @@
+= Using Config Store Yaml
+
+This extension provides support for loading stargate configuration from a YAML file.
+
+== YAML File Location
+
+Before the `ConfigStoreYaml` is registered in the OSGi, it will try to lookup the stargate config
+in the `/etc/stargate/stargate-config.yaml` location. Please note that the name of the actual YAML file is `stargate-config.yaml`
+If it does not exists, the error log is produced, and `ConfigStoreYaml` is not registered.
+You can modify the location where the config store lookup for config using the `stargate.config_store.yaml.location` system property that takes the absolute path to the stargate config YAML file.
+
+== YAML Config Format
+
+The YAML file used to load the Stargate configuration should have the following format:
+
+[source,yaml]
+----
+extension-1:
+  a: 1
+  b: "value"
+
+extension-2:
+  a: 2
+  b: "value_2"
+----
+
+Every extension should have its own dedicated section.
+When client is using the `ConfigStore#getConfigForExtension(String extensionName)` method to retrieve the config for a given extension,
+the `ConfigStoreYaml` loads only a dedicated section.
```

Comment:
I need to check the impl but why not load the whole file at startup and just create a map of maps where the key is the name of the component?

Reply:
We are doing just that. This API is returning an nested map for a given component to provide isolation between extensions from the client perspective.